<!-- slam.launch -->
<launch>

  <!--  ファイル名を指定するなら下記のargを変更すること -->
  <arg name="map_file" default="$(find target_scan_follower)/map/map.yaml"/><!-- raw map for amcl-->
  <arg name="map_movebase_file" default="$(find target_scan_follower)/map/map_movebase.yaml"/><!-- fixed map for move_base -->
  <arg name="map_movebase_topic_name" default="map_movebase"/>
  <arg name="location_file" default="$(find target_scan_follower)/map/map_location.yaml"/>
  <arg name="initial_pose_x" default="0.0"/>
  <arg name="initial_pose_y" default="0.0"/>
  <arg name="initial_pose_a" default="0.0"/>
  <arg name="custom_param_file" default="$(find target_scan_follower)/param/costmap_common_params.yaml"/>
  <arg name="waypoint_config_file" default="$(find target_scan_follower)/param/waypoint_nav.yaml"/>

  <!-- Map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <!-- Map_move_base server -->
  <node name="map_movebase_server" pkg="map_server" type="map_server" args="$(arg map_movebase_file)" >
    <remap from="map" to="$(arg map_movebase_topic_name)" />
    <remap from="static_map" to="static_map_movebase" />
  </node>


  <!-- AMCL -->
  <include file="$(find target_scan_follower)/launch/includes_turtlebot/amcl.launch.xml">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>

  <!-- Move_Base -->
  <include file="$(find target_scan_follower)/launch/includes_turtlebot/move_base.launch.xml">
    <arg name="custom_param_file" value="$(arg custom_param_file)"/>
    <arg name="map_topic" value="$(arg map_movebase_topic_name)"/>
  </include>

  <!-- waypoint_nav -->
  <group ns = "target_scan_follower">
    <param name="location_yaml_path" type="string" value="$(arg location_file)"/>
    <rosparam command="load" file="$(arg waypoint_config_file)"/>
    <node pkg="sobit_navigation" type="waypoint_nav" name="waypoint_nav" output="screen"/>
  </group>



</launch>
